packages/manager/src/components/Feedback/Feedback.tsx

```diff
@@ -1,42 +1,53 @@
-import React, { useEffect, Suspense } from 'react';
+import React, { useEffect, Suspense, useRef } from 'react';
 import { useSession } from 'next-auth/react';
-import { useCreateFeedback } from '@app/services/feedback';
 import { FeedbackInput } from '@app/services/feedback/types';
+import { createFeedback, useCreateFeedback } from '@app/services/feedback/queries';
+import { toast } from 'react-hot-toast';
+import { AxiosError } from 'axios';
 
 export const Feedback = () => {
   const { data: session } = useSession();
+  // const createFeedback = useCreateFeedback();
   const createFeedback = useCreateFeedback();
+
   useEffect(() => {
     const loadOpcFeedback = async () => {
       await import('@one-platform/opc-feedback/dist/opc-feedback');
     };
-
     const feedback = document.querySelector('#opc-feedback');
-
     if (feedback) {
       loadOpcFeedback();
+
       feedback.addEventListener('opc-feedback:submit', async (event: any) => {
-        const stackInfoData = { ...event.detail.data.stackInfo };
-        stackInfoData.path = window.location.pathname;
-        const username = session?.user?.email ? session?.user?.email.split('@')[0] : '';
+        if (event.detail.submitted) {
+          // If already submitted, do nothing
+          return;
+        }
```

Comment:
Why this if loop

Reply:
i am using this as a flag

Reply:
Add a TODO in Code about he root cause of this code.